devices/Hx711/Hx711.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Iot.Device.Hx711
+{
+    /// <summary>
+    /// Scale class proving a wrapper to the Hx711 to be used as a scale for weight measurements.
+    /// </summary>
+    public class Scale
+    {
+        // number of clock periods to clock out in order to read a sample
+        private const int ClockPeriodsForSampleReading = 24;
+
+        // pulse train required to read a sample and setup gain factor for next reading
+        private byte[] readSamplePulseTrain;
+
+        private readonly bool[] _rawSample = new bool[24];
+        private readonly SpiDevice _spiDevice;
+
+        /// <summary>
+        /// Gets or sets the value that's subtracted from the actual reading.
+        /// </summary>
+        public int Offset { get; set; }
+
+        /// <summary>
+        /// Gets or sets the gain factor that the Hx711 uses when sampling.
+        /// </summary>
+        /// <remarks>
+        /// The default value is <see cref="GainLevel.Gain_128"/>.
+        /// </remarks>
+        public GainLevel Gain { get; set; }
+
+        /// <summary>
+        /// Gets or sets the number of samples that will be taken and then averaged when performing a <see cref="Read"/> operation.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// The default value is 3 samples.
+        /// </para>
+        /// <para>
+        /// The number of samples to take has impact on the time it takes for a <see cref="Read"/> operation to complete. The Hx711, in the current configuration, as sample rate of 10 samples per second.
+        /// </para>
+        /// </remarks>
+        public uint SampleAveraging { get; set; } = 3;
```

Comment:
is 0 an appropriate value? Any unauthorized number? If yes, you may want to review this one to check the value

Reply:
No, you have to have at least 1 sample. That's why I've used an unsigned (already takes care of not using negative values).
I'll add the validation for 0.